File: src/ui/asq-ui/src/app/_models/domain/user-dto.ts
import { BaseDomainDto } from './base-domain-dto';
import { AccountTypeEnu } from './account-type-enu';
import { FocusUserMappingDto } from './focus-user-mapping-dto';
import { HostDto } from './host-dto';
import { ImgDto } from './img-dto';
import { PaymentInfoDto } from './payment-info-dto';

export class UserDto extends BaseDomainDto{
    public username: string = null;
    public accountType: AccountTypeEnu = AccountTypeEnu.STUDENT;
    public password: string = null;
    public email: string = null;
    public name: string = null;
    public surname: string = null;
    public paymentInfoId: number | null = null;
    public hostId: number | null = 0;
    public imgId: number = 0;
    public timezoneId: number = 0;
    public interests: FocusUserMappingDto[] = null;

    public host: HostDto = null;
    public paymentInfo: PaymentInfoDto = null;
    public img: ImgDto = null;

    //public passwordConfirm: string = null;
    public token: string = null;
    public isHost: boolean = false;
    public isContractSigned: boolean = null;

    constructor(){
        super();
    }
}

File: src/lib/models/DTO/Domain/UserDto.cs
using System;
using System.Collections.Generic;
using extensions;

namespace DTO.Domain
{
    public class UserDto: BaseDomainDto
    {
        public string Username{ get; set; }
        public string Password{ get; set; }
        public string Email{ get; set; }
        public string Name{ get; set; }
        public string Surname{ get; set; }
        public long? PaymentInfoId {get; set; }
        public long? HostId {get; set; }
        public long ImgId {get; set; }
        public int TimezoneId {get; set; }
        public AccountTypeEnu AccountType { get; set; }

        public PaymentInfoDto PaymentInfo{ get; set; }
        public HostDto Host{ get; set; }
        public ImgDto Img{ get; set; }
        public TimezoneDto Timezone{ get; set; }
        public ICollection<FocusUserMappingDto> Interests{ get; set; }
        public string Token{ get; set; }

        new public bool Validate(bool throwException = true)
        {
            base.Validate(throwException);

            bool invalid = this.Username.IsNullOrEmpty()
                || this.Password.IsNullOrEmpty()
                || this.Name.IsNullOrEmpty()
                || this.Surname.IsNullOrEmpty()
                || (this.Interests == null || this.Interests.Count == 0)
                || this.Img == null
                || this.TimezoneId <= 0;

            if(invalid && throwException)
            {
                throw new Exception("Invalid user structure");
            }

            return invalid;
        }
    }
}